R/importSNV.TCGA.R

```diff
@@ -0,0 +1,268 @@
+
+            # Compile function call record
+            if (missing(rSNV)) {
+                myCall[1] <- "importSNV.TCGA("
+                myCall[2] <- sprintf("fMAF = \"%s\", ", fMAF)
+                myCall[3] <- sprintf("silent = %s, ", as.character(silent))
+                myCall[4] <- sprintf("writeLog = %s, ", as.character(writeLog))
+                myCall[5] <- sprintf("na.rm = %s)", as.character(na.rm))
+                myCall <- paste0(myCall, collapse = "")
+            } else {
+                myCall[1] <- "importSNV.TCGA("
+                myCall[2] <- sprintf("fMAF = \"%s\", ", fMAF)
+                myCall[3] <- sprintf("rSNV = \"%s\", ", rSNVu)
+                myCall[4] <- sprintf("silent = %s, ", as.character(silent))
+                myCall[5] <- sprintf("writeLog = %s, ", as.character(writeLog))
+                myCall[6] <- sprintf("na.rm = %s)", as.character(na.rm))
+                myCall <- paste0(myCall, collapse = "")
+            }
+
+            # indicate input object name(s)
+            #   (NA for this importSNV.TCGA())
+
+            # Record progress information
+            myNotes <- c(myNotes, paste("Total number of files in input fMAF -", length(fMAF)))
+            myNotes <- c(myNotes, paste("Files successfully processed -", filesFinished))
+            myNotes <- c(myNotes, paste("Files dropped due to incomptibility -", numFilesInvalid))
+
+            # send info to log file
+            logEvent(eventTitle = myTitle,
+                      eventCall = myCall,
+                      notes = myNotes)
+        }
+    }
+}
+
+.mafHeaderCheck <- function(file) {
+    headValid <- character()
+    validMAFheader <- c("Hugo_Symbol", "Chromosome", "Start_Position", "End_Position",
+                        "Strand", "Variant_Classification", "Variant_Type", "Reference_Allele",
+                        "Tumor_Seq_Allele1", "Tumor_Seq_Allele2", "Tumor_Sample_Barcode",
+                        "Tumor_Sample_UUID")
+    if (! all(read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                     na.strings = " ", nrows=1)[,c(1,5,6,7,8,9,10,11,12,13,16,33)]
+        == validMAFheader)) {
+        headValid <- c(headValid, "Header of input MAF file did not match the requirements.")
+    }
+    return(headValid)
+}
+
+.rSNVheadercheck <- function(file) {
+    headValid <- character()
+    validSNVheader <- c("sym", "chr", "start", "end","strand", "class", "type", "aRef",
+                        "a1", "a2", "TCGA","UUID")
+    if (! all(read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                     na.strings = " ", nrows=1)[,c(1,2,3,4,5,6,7,8,9,10,11,12)]
+        == validSNVheader)) {
+        headValid <- c(headValid, "Header of input rSNV file did not match the requirements.")
+    }
+    return(headValid)
+}
+
+.fileValidity <- function(file) {
+    validExt <- character()
+    if (! (strsplit(file, "\\.")[[1]][-1] %in% c("maf", "txt", "tsv"))) {
```

Comment:
This will fail if there is more than one period in the filename - which is a perfectly legal name. You need to extract only the extension. And don't use "file" when you mean the filename, and ```file()``` is a function anyway and is therefor not a good choice of variable name. 
```
regmatches(fName, regexpr("[^.]+$", fName))
```